File: src/main/java/com/wxl/dyttcrawler/config/DyttWebMvcConfiguration.java
package com.wxl.dyttcrawler.config;

import com.wxl.dyttcrawler.web.dto.AggOneResult;
import com.wxl.dyttcrawler.web.dto.serializer.AggResultSerializer;
import com.wxl.dyttcrawler.web.error.DyttErrorAttributes;
import org.springframework.boot.autoconfigure.jackson.Jackson2ObjectMapperBuilderCustomizer;
import org.springframework.boot.web.servlet.error.DefaultErrorAttributes;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

/**
 * Create by wuxingle on 2020/7/12
 * web配置
 */
@Configuration
public class DyttWebMvcConfiguration {


    @Bean
    public DefaultErrorAttributes errorAttributes() {
        return new DyttErrorAttributes();
    }

    @Bean
    public Jackson2ObjectMapperBuilderCustomizer jacksonSerializerCustomizer() {
        return builder -> {
            builder.serializerByType(AggOneResult.class, new AggResultSerializer());
        };
    }


}

File: src/main/kotlin/com/wxl/dyttcrawler/config/DyttWebMvcConfiguration.kt
package com.wxl.dyttcrawler.config

import com.wxl.dyttcrawler.web.error.DyttErrorAttributes
import org.springframework.boot.autoconfigure.jackson.Jackson2ObjectMapperBuilderCustomizer
import org.springframework.boot.web.servlet.error.DefaultErrorAttributes
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

/**
 * Create by wuxingle on 2021/10/11
 * web配置
 */
@Configuration
class DyttWebMvcConfiguration {

    @Bean
    fun errorAttributes(): DefaultErrorAttributes {
        return DyttErrorAttributes()
    }

    @Bean
    fun jacksonSerializerCustomizer(): Jackson2ObjectMapperBuilderCustomizer {
        return Jackson2ObjectMapperBuilderCustomizer {
        }
    }
}
